src/Error/ExceptionRenderer.php

```diff
@@ -90,6 +95,26 @@ class ExceptionRenderer implements ExceptionRendererInterface
      */
     protected $request;
 
+    /**
+     * Map of exceptions to http status codes.
+     *
+     * This can be customized for users that don't want specific exceptions to throw 404 errors
+     * or want their application exceptions to be automatically converted.
+     *
+     * @var array
+     * @psalm-var array<string, int>
+     */
+    protected $exceptionHttpCodes = [
+        // Controller exceptions
+        MissingActionException::class => 404,
+        // Datasource exceptions
+        InvalidPrimaryKeyException::class => 404,
+        PageOutOfBoundsException::class => 404,
+        RecordNotFoundException::class => 404,
+        // Http exceptions
+        MissingControllerException::class => 404,
```

Comment:
Instead of needing this I think we should just make `MissingControllerException` extend `HttpException`. It's a bit weird that it's under `Http/Exception/` and extends `Core`Exception`Exception` instead of `Http\ExceptionHttpException`.

Reply:
Changing the parent might have been missed when it moved from under `Routing/Exception/`.

Reply:
The custom error template wont render. Maybe we can figure that out.

Reply:
Yeah we could a add a check in `_template()` method to use the special template when debug is on.

Reply:
We'll have to figure that out separate because it explicitly doesn't render http exceptions in debug.